swApiClient.cpp

```diff
@@ -1,10 +1,27 @@
-#include "swApiClient.h"
+#include <stdlib.h>
+#include <string.h>
 #include <iostream>
+#include "swApiClient.h"
 
+using namespace std;
+using json = nlohmann::json;
 
-/* static member init */
-std::string swApiClient::BASE_URL = "http://api.shapeways.com";
-std::string swApiClient::API_VERSION = "v1";
+/* Helper Functions */
+struct curl_slist * swApiClient::SetAuthHeader() {
+  char *headerStr = (char*)malloc(250);
+  strcpy(headerStr, "Authorization: Bearer ");
+  strcat(headerStr, this->accessToken.c_str());
+  cout << headerStr << endl;
+  struct curl_slist *headers = NULL;
+  headers = curl_slist_append(headers, headerStr);
+  return headers;
+}
+
+static size_t WriteCallback(void *contents, size_t size, size_t nmemb, void *userp)
```

Comment:
maybe expand nmemb to number_of_elements or something to improve readability. same for userp

Reply:
yeah fair, it's a reference client: it should be readable